assets/js/components/GoogleChart/index.js

```diff
@@ -197,6 +212,114 @@ export default function GoogleChart( props ) {
 		}
 	}, [ onMouseOver, onMouseOut ] );
 
+	/**
+	 * Checks to see if the date is within the date range.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Date} date The date to check.
+	 * @return {boolean}  `true` if the date is within the current date
+	 *                    range, `false` otherwise.
+	 */
+	const isDateWithinRange = useCallback(
+		( date ) => {
+			// If any of the dates are not set, return false.
+			if ( ! date || ! startDate || ! endDate ) {
+				return false;
+			}
+
+			if (
+				// Don't render this marker if the date is before the first date
+				// in the current date range.
+				date.getTime() < stringToDate( startDate ).getTime() ||
+				// Don't render this marker if the date is after the last date
+				// in the current date range.
+				date.getTime() > stringToDate( endDate ).getTime()
+			) {
+				return false;
+			}
+
+			return true;
+		},
+		[ startDate, endDate ]
+	);
+
```

Comment:
I'd have thought it would be slightly easier to manage taking the approach of creating a new filtered version of `dateMarkers` and then using that in place of it in the rest of the file, that would avoid having to check `isDateWithinRange()` in two places and return early in the respective loops.

Not sure if there's some reason _not_ to do this that I haven't thought of - have you already considered this approach?

Reply:
Oh, shoot, that's totally fair. I just blanked on that, but I'll make the change 👍🏻